doc/sphinx/conf.py

```diff
@@ -140,7 +140,10 @@ def setup(app):
 
 # If true, SmartyPants will be used to convert quotes and dashes to
 # typographically correct entities.
-#html_use_smartypants = True
+# Better to disable this to prevent alteration of command-line arguments
+# Sphinx <1.6 use `html_use_smartypants` and >=1.6 use `smartquotes`
+html_use_smartypants = False
```

Comment:
In another project to be compatible with both older and newer versions of Sphinx I've used:

```suggestion
from sphinx import __version__ as sphinx_version

sphinx_version_parts = [int(i) for i in sphinx_version.split('.')]
if sphinx_version_parts[0] == 1 and sphinx_version_parts[1] < 6:
    html_use_smartypants = False
```

I didn't had to set `smartquotes` to `False`, but I use sphinx-argparse for documenting the CLI arguments.

My 2 cents

Reply:
Thanks for the suggestion, I have rewritten the commit to make use of the test code you propose.